lib/tests/tests.c

```diff
@@ -973,8 +974,23 @@ test_demographic_events(void)
 
         CU_ASSERT_EQUAL(msp_debug_demography(&msp, &time), MSP_ERR_BAD_STATE);
 
-        ret = msp_initialise(&msp);
-        CU_ASSERT_EQUAL(ret, 0);
+        if (model == 0) {
+            ret = msp_initialise(&msp);
+            CU_ASSERT_EQUAL(ret, 0);
+        } else {
+            /* Check row sum constraints for migration matrix in DTWF */
+            ret = msp_set_migration_matrix(&msp, 4, dtwf_invalid_migration_matrix);
+            CU_ASSERT_EQUAL(ret, 0);
+            ret = msp_initialise(&msp);
+            CU_ASSERT_EQUAL(ret, 0);
+            ret = msp_run(&msp, DBL_MAX, ULONG_MAX);
+            CU_ASSERT_EQUAL(ret, MSP_ERR_DTWF_MIGRATION_MATRIX_NOT_STOCHASTIC);
+            /* Reset to valid state */
+            ret = msp_set_migration_matrix(&msp, 4, migration_matrix);
+            CU_ASSERT_EQUAL(ret, 0);
+            ret = msp_reset(&msp);
+            CU_ASSERT_EQUAL(ret, 0);
+        }
```

Comment:
Gee, I would have pulled this out into a separate test, but testing everything all at once seems to be how we're rolling in this file already, so I'll let Jerome weigh in on this if he wants. LGTM otherwise.

Reply:
Yes, I agree with Peter - this test is already badly factored and too complicated, so let's pull this new check out into it's own test,  ``test_dtwf_migration_rates``. I would use  ``test_dtwf_zero_pop_size`` as inspiration here (i.e., copy most of it). It's annoying there's so much copying and duplication in this file, but hopefully we can split it up a bit soon (#1122) and that'll make things easier to refactor.